BIOM-LD
=======

Summary
-------

This is a first hacky attempt towards a full mapping of the BIOM format to a more Linked Data friendly JSON-LD format, and beyond. 

Rationale
---------

The [Biological Observation Matrix (BIOM) format](http://biom-format.org/) is a [JSON](http://en.wikipedia.org/wiki/JSON) based specification to represent biological observation tables. For example, it is used in the [Earth Microbiome Project](http://www.earthmicrobiome.org/) for storing environmental metagenomics data in a common an efficient format. On the other hand, [JSON-LD](http://json-ld.org/) is also a JSON-based representation of [Linked Data](http://en.wikipedia.org/wiki/Linked_data) resources, with the status of [W3C recommendation](http://www.w3.org/standards/faq#std). Thus, one can map the BIOM format to BIOM-LD (from plain JSON to JSON-LD) and obtain a Linked Data representation that can be directly plugged into the [web of data](http://lod-cloud.net/).  

Moving to a Linked Data representation (Not necessarily based in JSON-LD, see bellow) over a pure JSON format has the following advantages, specially from the publication/interoperability point of view (but not from the efficient storage point of view):

* Publishing our data in Linked Data means that other datasets can be linked to ours (i.e. our dataset becomes more "discoverable" over the web) or we can link our dataset to other datasets and integrate information easily. 
* Since we are using [RDF](http://www.w3.org/standards/techs/rdf#w3c_all), it is easy to merge our datasets with other datasets. This is specially interesting if common vocabularies like EnvO or NCBI taxonomy are used to represent row and column metadata, and/or [SPARQL federated queries](http://www.w3.org/TR/sparql11-federated-query/) are used.
* Since the BIOM specification is represented as an [OWL](http://www.w3.org/standards/techs/owl#w3c_all) ontology, the specification, rather than being pure text, becomes computationally explicit: programs that consume BIOM data can be more easily written, reasoning can be used to check validity, specific validators (e.g. for metadata, value ranges, ... ) can be more easily written, and in general any programmatic endeavour becomes easier and more maintainable in the long term.

The mapping
-----------

The mapping is provided as a concrete example, a first prototype, through the files listed bellow. A BIOM file is basically a table with some metadata attached to it: in this mapping, it becomes and instance that is a member of any of the subclasses of `biom:Table`, for example `biom:OTUTable`. The information about the table (generated by, table type, etc.) is translated to triples whose subject is the table instance. Each row of the table is translated to a [named graph](http://en.wikipedia.org/wiki/Named_graph) in which each column is linked to a cell in a triple. The files:

* `biom-format.owl`: this represents the [BIOM specification](http://biom-format.org/documentation/format_versions/biom-1.0.html) in a machine and web friendly way.
* `min_dense_otu_table.biom`: the example BIOM file used for the mapping.
* `min_dense_otu_table.biom-ld`: the translation of `min_dense_otu_table.biom` into JSON-LD.
* `min_dense_otu_table.biom-ld.n-qads`: `min_dense_otu_table.biom-ld` in n-quads format (from [JSON-LD validator](http://json-ld.org/playground/)).

The mapping is provided as JSON-LD to retain JSON compatilibity with any tools that consume BIOM. However, since the BIOM specification is already codified in an ontology, it makes more sense, if one is going to publish large BIOM files as Linked Data, to write a program that consumes BIOM files and injects triples directly to a [triple store](http://en.wikipedia.org/wiki/Triplestore), using the BIOM-format ontology provided here as a guide.

See issues for things that need to be done (a lot!).



